src/formatDate.js

```diff
@@ -50,7 +50,88 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateElements = date.split(fromFormat[fromFormat.length - 1]);
+  const separator = toFormat[toFormat.length - 1];
+  const resultingDate = [];
+
+  let yearFromIndex = 0;
+  let monthFromIndex = 0;
+  let dayFromIndex = 0;
+  let yearToIndex = 0;
+  let monthToIndex = 0;
+  let dayToIndex = 0;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YY':
+        yearFromIndex = i;
+        break;
+
+      case 'YYYY':
+        yearFromIndex = i;
+        break;
+
+      case 'MM':
+        monthFromIndex = i;
+        break;
+
+      case 'DD':
+        dayFromIndex = i;
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YY':
+        yearToIndex = i;
+        break;
+
+      case 'YYYY':
+        yearToIndex = i;
+        break;
+
+      case 'MM':
+        monthToIndex = i;
+        break;
+
+      case 'DD':
+        dayToIndex = i;
+        break;
+    }
```

Comment:
Here you can fill `resultingDate` array without `switch...case`, only by pushing values, f.e. `dateObject[toFormat[i]]`.